ACM-18012: Add `objectSelector` doc (GRC) (#7527)
open-cluster-management/rhacm-docs: governance/adv_template_process.adoc

If you need to manually initiate an update, use the special annotation, `policy.open-cluster-management.io/trigger-update`, to indicate changes for the data referenced by the templates. Any change to the special annotation value automatically initiates template processing. Additionally, the latest contents of the referenced resource are read and updated in the policy definition that is propagated for processing on managed clusters. A way to use this annotation is to increment the value by one each time.

[#object-selector]
== Configuration policy `objectSelector`

> To iterate over many objects filtered by a label with `ConfigurationPolicy`, use the `objectSelector` and specify the `kind` in the `objectDefinition`. The `objectSelector` is similar to the `namespaceSelector`, except that it does not have name filtering capabilities. To filter by name, use the `skipObject` Go template function. If you reference `{{ skipObject }}` at any time, it signals to skip that particular object from those selected by the `objectSelector` for the policy.

Reviewer: ```suggestion
To iterate over many objects that are filtered by a label with `ConfigurationPolicy`, use the `objectSelector` and specify the `kind` in the `objectDefinition`. The `objectSelector` is similar to the `namespaceSelector`, except that it does not have name filtering capabilities. To filter by name, use the `skipObject` Go template function. If you reference `{{ skipObject }}` at any time, it signals to skip that particular object from those that are selected by the `objectSelector` for the policy.
```

> For an example, view the following YAML sample that selects objects that have the label `foo: bar` but skips objects that have a name with the suffix `-prod`:

Reviewer: ```suggestion
See the following YAML sample that selects objects that have the label `foo: bar`, but skips objects that have a name with the suffix `-prod`:
```

[#object-selector]
== Configuration policy `objectSelector`

> To iterate over many objects that use a label selector with `ConfigurationPolicy`, use the `objectSelector` and specify the `kind` in the `objectDefinition`. The `objectSelector` is similar to the `namespaceSelector`, except that it does not automatically have name filtering capabilities. To filter by name, use the `skipObject` function. If you reference `{{ skipObject }}` at any time inside the Go template, it signals the `objectSelector` to skip that particular object for the policy.

Author: The `objectSelector` is a label selector that selects/filters objects by label.
I'm also not sure "automatically" makes sense to me for `skipObject` since it's an additional step to use in addition to the `objectSelector`.
Given that, would this wording make sense?

```suggestion
To iterate over many objects filtered by a label with `ConfigurationPolicy`, use the `objectSelector` and specify the `kind` in the `objectDefinition`. The `objectSelector` is similar to the `namespaceSelector`, except that it does not have name filtering capabilities. To filter by name, use the `skipObject` Go template function. If you reference `{{ skipObject }}` at any time, it signals to skip that particular object from those selected by the `objectSelector` for the policy.
```
